### PR TITLE
Cmake improvements for ITTAPI

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
Moreover:
- Remove setting `-g` which is already default for `Debug` build type
```
CMAKE_CXX_FLAGS_DEBUG:STRING=-g
CMAKE_CXX_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
CMAKE_C_FLAGS_DEBUG:STRING=-g
CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
```
- Remove setting `NDEBUG` which is already set for non-debug:
```
CMAKE_C_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG
CMAKE_C_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
```
- Removed `_DEBUG` definition since it's not really used by the project
- Used makes variable `CMAKE_DL_LIBS` instead of manual linking with `dl`
- Removing cmake's condition for release / debug allows to support multi-config generators like MSVC, Ninja Multi-Config, Xcode and others.

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
@abrown could you please review?

**Review by abrown — APPROVED** [review]
LGTM but I'll let @ichepiga or @elena.nuzhnova take a look since they typically review the C side of all of this.

File: CMakeLists.txt [hunk 1]
```diff
@@ -26,18 +26,14 @@ if(CMAKE_SIZEOF_VOID_P MATCHES "8" AND NOT(FORCE_32))
...
-else()
-    if (NOT WIN32)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2")
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2")
-    endif()
-    add_definitions(-DNDEBUG)
+if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_CXX_COMPILER_ID MATCHES "^(Apple)?Clang$")
+    # override default -O3
+    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O2")
+    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O2")
```
Comment (ekovanova) [inline CMakeLists.txt:32]: To make CMAKE_C_FLAGS_RELEASE and CMAKE_CXX_FLAGS_RELEASE work could you please add the RELEASE type in buildall.py:
```
-            ("-DCMAKE_BUILD_TYPE=Debug" if args.debug else ""),
+            ("-DCMAKE_BUILD_TYPE=Debug" if args.debug else "-DCMAKE_BUILD_TYPE=RELEASE"),
```

Comment (ekovanova) [reply CMakeLists.txt:32]: Does it really overrides the -O3 flag?
    # override default -O3
    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O2")
    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O2")
    
I just tried to build release libs and it looks like we have both flags (-O3 and -O2) now:
  
 ```
[ 60%] Building C object CMakeFiles/ittnotify.dir/src/ittnotify/ittnotify_static.c.o
/usr/bin/cc  -I/localdisk/ittapi_git/ittapi/include -I/localdisk/ittapi_git/ittapi/src/ittnotify  -O3 -DNDEBUG -O2 -fPIC   -o CMakeFiles/ittnotify.dir/src/ittnotify/ittnotify_static.c.o   -c /localdisk/ittapi_git/ittapi/src/ittnotify/ittnotify_static.c
[ 80%] Building C object CMakeFiles/ittnotify.dir/src/ittnotify/jitprofiling.c.o
/usr/bin/cc  -I/localdisk/ittapi_git/ittapi/include -I/localdisk/ittapi_git/ittapi/src/ittnotify  -O3 -DNDEBUG -O2 -fPIC   -o CMakeFiles/ittnotify.dir/src/ittnotify/jitprofiling.c.o   -c /localdisk/ittapi_git/ittapi/src/ittnotify/jitprofiling.c
```

Comment (ekovanova) [reply CMakeLists.txt:32]: @ilya-lavrenov thank you much for the detailed description for the change!
now we just need to make RELEASE flags work and override the -O3 flag with -O2

Comment (ilya-lavrenov) [reply CMakeLists.txt]: Fixed by adding `Release` as a default configuration.

**Review by ekovanova — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -11,41 +11,47 @@ if (POLICY CMP0048)
...
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g")
-    endif()
-else()
-    if (NOT WIN32)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2")
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2")
-    endif()
-    add_definitions(-DNDEBUG)
+if(CMAKE_COMPILER_IS_GNUCCC OR CMAKE_CXX_COMPILER_ID MATCHES "^(Apple)?Clang$")
```
Comment (ilya-lavrenov) [inline CMakeLists.txt:39]:
```suggestion
if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "^(Apple)?Clang$")
```

**Pull request merged**